src/wrappers/koa/KoaWrapper.ts

```diff
@@ -0,0 +1,101 @@
+import {ApplicationManager} from '../../application/ApplicationManager';
+import {ClassNames, DomainNames} from '../../Constants';
+import WrapperUtils from '../WebWrapperUtils';
+import ConfigProvider from '../../config/ConfigProvider';
+import Reporter from '../../Reporter';
+import * as KoaExecutor from './KoaExecutor';
+import ThundraLogger from '../../ThundraLogger';
+import ExecutionContextManager from '../../context/ExecutionContextManager';
+import ExecutionContext from '../../context/ExecutionContext';
+import LambdaUtils from '../../utils/LambdaUtils';
+import ModuleUtils from '../../utils/ModuleUtils';
+import Utils from '../../utils/Utils';
+
+export function koaMiddleWare(opts: any = {}) {
+    ThundraLogger.debug('<KoaWrapper> koaMiddleWare running ...');
+
+    ApplicationManager.setApplicationInfoProvider().update({
```

Comment:
Also, is it possible to move the following logic into `WebWrapperUtils`, take `applicationClassName` as param and call it from there and then return reporter and plugins here:

```js
    ApplicationManager.setApplicationInfoProvider().update({
        applicationClassName: ClassNames.KOA,
        applicationDomainName: DomainNames.API,
    });

    const appInfo = ApplicationManager.getApplicationInfo();
    ApplicationManager.getApplicationInfoProvider().update({
        applicationId: WrapperUtils.getDefaultApplicationId(appInfo),
    });

    const config = opts.config || ConfigProvider.thundraConfig;
    const {apiKey} = config;
    const reporter = opts.reporter || new Reporter(apiKey);
    const pluginContext = opts.pluginContext || WrapperUtils.createPluginContext(apiKey, KoaExecutor);
    const plugins = opts.plugins || WrapperUtils.createPlugins(config, pluginContext);

    if (!opts.disableAsyncContextManager) {
        WrapperUtils.initAsyncContextManager();
    }
```